libs/bArray.lua

```diff
@@ -146,8 +156,8 @@ end
 byteArray.read8 = function(self, quantity)
 	quantity = quantity or 1
 
-	local byteStack = table_arrayRange(self.stack, 1, quantity)
-	self.stack = table_arrayRange(self.stack, quantity + 1)
+	local byteStack = table_arrayRange(self.stack, self.pos, self.pos + quantity)
+	self.pos = self.pos + quantity
```

Comment:
By not refreshing stack's elements you risk to make the table way too huge, consuming unnecessary memory.